### ISSUE TITLE
Rubocop destroyes my ApplicationController

### ISSUE THREAD

**Issue by nleo** [bbatsov/rubocop#5584]
Was:

    class ApplicationController < ActionController::Base
      include Pundit
      protect_from_forgery with: :exception
      before_action :authenticate_user!
      before_action :otp_required_check, if: ->{OTP_ENABLED}
      before_action :configure_permitted_parameters, if: :devise_controller?
      before_action :set_paper_trail_whodunnit

Became:

    class ApplicationController < ActionControl
    }        e_action :otp_required_check, if: -> { OTP_ENABLED }
      before_action :configure_permitted_parameters, if: :devise_controller?
      before_action :set_paper_trail_whodunnit

    $ rubocop -V
    0.52.1 (using Parser 2.4.0.2, running on ruby 2.4.3 x86_64-linux-gnu)

    $ cat .rubocop.yml 
    AllCops:
      Exclude:
        - 'db/seeds.rb'
        - 'db/schema.rb'
        - 'db/migrate/*'
        - config/initializers/devise.rb
        - bin/yarn
        - bin/update
        - bin/setup
    Style/AsciiComments:
      Enabled: false
    Documentation:
      Enabled: false
    Style/LambdaCall:
      Enabled: false
    Style/MultilineBlockChain:
      Enabled: false
    Style/ClassAndModuleChildren:
      Enabled: false
    Style/FrozenStringLiteralComment:
      Enabled: false
    Bundler/OrderedGems:
      Enabled: false
    Style/SymbolArray:
      Enabled: false
    Layout/SpaceInsideArrayLiteralBrackets:
      Enabled: false
    Metrics/LineLength:
      Max: 120
    Metrics/MethodLength:
      Max: 30

**Comment by nleo** [comment]
One more strage thing

    def index
      @items = Item.all
    -    authorize @items, :index?
    +    authorize @items, :index
    +}

**Comment by bquorning** [comment]
I cannot reproduce this. What output would you get from running `rubocop --format offenses` ?

**Comment by nleo** [comment]
```
$ rubocop --format offenses app/controllers/application_controller.rb 

1  Layout/BlockEndNewline
1  Layout/IndentationConsistency
1  Lint/UselessAssignment
1  Metrics/AbcSize
1  Style/GuardClause
--
5  Total
```

```
$ rubocop -a app/controllers/application_controller.rb 
Inspecting 1 file
E

Offenses:

app/controllers/application_controller.rb:6:10: E: Lint/Syntax: unexpected token tIDENTIFIER
(Using Ruby 2.4 parser; configure using TargetRubyVersion parameter, under AllCops)
end      on :otp_required_check, if: -> { OTP_ENABLED }
         ^^
app/controllers/application_controller.rb:19:7: C: [Corrected] Layout/IndentationConsistency: Inconsistent indentation detected.
      redirect_to(request.referrer || root_path) end
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/controllers/application_controller.rb:19:50: C: [Corrected] Layout/BlockEndNewline: Expression at 19, 50 should be on its own line.
      redirect_to(request.referrer || root_path) end
                                                 ^^^
app/controllers/application_controller.rb:34:1: E: Lint/Syntax: unexpected token kEND
(Using Ruby 2.4 parser; configure using TargetRubyVersion parameter, under AllCops)
end
^^^

1 file inspected, 4 offenses detected, 2 offenses corrected
```

now it makes another change, no like before

```
   before_action :authenticate_user!
-  before_action :otp_required_check, if: -> { OTP_ENABLED }
+  before_
+end      on :otp_required_check, if: -> { OTP_ENABLED }
   before_action :configure_permitted_parameters, if: :devise_controller?
```
oh, i find a problem

```
respond_to do |format|
      format.html do
          flash[:error] = t "policy.#{exception.query}", scope: 'pundit', default: :default
      redirect_to(request.referrer || root_path) end
      format.json { render json: { error: I18n.t('not_authorized') }, status: 403 }
    end
```

I fix it manualy and it stops destroy code in head of conrotller

**Comment by nleo** [comment]
second one with `:index?`

```
2  Metrics/AbcSize
1  Layout/AlignHash
1  Layout/BlockEndNewline
1  Metrics/LineLength
1  Style/BlockDelimiters
1  Style/NumericPredicate
--
7  Total
```

**Comment by mikegee** [comment]
Can you provide the minimal ruby file that produces the problem?

**Comment by nleo** [comment]
@mikegee https://github.com/nleo/rubocop-bug - have 2 commits - initial state and after `rubocop -a`

**Comment by bquorning** [comment]
Ah, I see it now. It’s `BlockEndNewline` acting up again.

Duplicate of #5336 and #5232.

**Comment by bquorning** [comment]
I believe this issue has been fixed by #5590.
